framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -3549,6 +3605,26 @@ protected void addAliasedTypeAnnotation(
         aliases.put(aliasName, new Alias(aliasName, canonicalAnno, false, null, null));
     }
 
+    /**
+     * Adds the annotation, whose fully-qualified name is given by {@code aliasName}, as an alias
+     * for the canonical annotation {@code canonicalAnno} that will be used by the Checker Framework
+     * in the alias's place.
+     *
+     * <p>Use this method if the alias class is not necessarily on the classpath at Checker
+     * Framework compile and run time. Otherwise, use {@link #addAliasedTypeAnnotation(Class,
+     * AnnotationMirror)} which prevents the possibility of a typo in the class name.
+     *
+     * @param aliasName the canonical name of the aliased annotation
+     * @param canonicalAnno the canonical annotation
+     */
+    // aliasName is annotated as @FullyQualifiedName because there is no way to confirm that the
+    // name of an external annotation is a canoncal name.
+    protected void addAliasedTypeAnnotation(
```

Comment:
Are these two additional overloads (`addAliasedTypeAnnotation` and `addAliasedDeclaAnnotation` below) really worth it?
Wouldn't it be easier to change `parseAliasesFromString` to create the `AnnotationMirror` from the class directly and return that in the pair?

Reply:
It's not needed for `addAliasedTypeAnnotation` but for `addAliasedDeclaAnnotation` it stores the FQN of the canonical qual instead of the `AnnotationMirror`. I will delete the overload for `addAliasedTypeAnnotation`.